src/adapter/src/explain/optimizer_trace.rs

```diff
@@ -91,6 +94,88 @@ impl OptimizerTrace {
         }
     }
 
+    /// Convert the optimizer trace into a vector or rows that can be returned
+    /// to the client.
+    pub fn into_rows(
```

Comment:
As suggested in [this comment](https://github.com/MaterializeInc/materialize/pull/24887#discussion_r1476149032).